rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryStore.scala

```diff
@@ -17,24 +17,27 @@ object HistoryStoreInstances {
   def historyStore[F[_]: Sync](store: Store[F]): HistoryStore[F] = new HistoryStore[F] {
     // TODO put list
     override def put(tries: List[Trie]): F[Unit] = {
-      val data = tries
-        .map { t =>
-          val key   = Trie.hash(t)
-          val bytes = Trie.codecTrie.encode(t).get
-          (key, bytes)
-        }
-      store.put(data)
+      def put(t: Trie): F[Unit] = {
+        val k = Trie.hash(t)
+        for {
+          b ← Trie.codecTrie.encode(t).get
+          _ ← store.put(k, b)
+        } yield ()
+      }
+      for {
+        _ ← tries traverse put
+      } yield ()
```

Comment:
I remember talking about this with you. is this the final version?

Reply:
issue with merge. will fix. tnx